presto-spi/src/main/java/com/facebook/presto/spi/ConnectorTableLayout.java

```diff
@@ -51,7 +55,8 @@ public ConnectorTableLayout(
             Optional<ConnectorNodePartitioning> nodePartitioning,
             Optional<Set<ColumnHandle>> streamPartitioningColumns,
             Optional<DiscretePredicates> discretePredicates,
-            List<LocalProperty<ColumnHandle>> localProperties)
+            List<LocalProperty<ColumnHandle>> localProperties,
+            TableStatistics statistics)
```

Comment:
I will follow @kokosing verbal suggestion to have separate `TableStatisticsProvider` which will be obtained from `Connector`. This way we do not need to compute statistics for cases when those are not needed by optimizer.